{"events":[{"defined_class":"com.example.accesscontrol.exception.GlobalExceptionHandlerTest$DatabaseIssues","event":"call","id":177737,"lineno":83,"method_id":"handleDatabaseIssue_dataAccessFailure","path":"src/test/java/com/example/accesscontrol/exception/GlobalExceptionHandlerTest.java","receiver":{"class":"com.example.accesscontrol.exception.GlobalExceptionHandlerTest$DatabaseIssues","object_id":2047363947,"value":"com.example.accesscontrol.exception.GlobalExceptionHandlerTest$DatabaseIssues@7a084b6b"},"static":false,"thread_id":1},{"defined_class":"com.example.accesscontrol.exception.GlobalExceptionHandler","event":"call","id":177738,"lineno":35,"method_id":"handleDatabaseConnectionIssue","parameters":[{"class":"org.springframework.dao.DataAccessResourceFailureException","kind":"req","name":"ex","object_id":94024704,"value":"org.springframework.dao.DataAccessResourceFailureException: db pool exhausted"}],"path":"src/main/java/com/example/accesscontrol/exception/GlobalExceptionHandler.java","receiver":{"class":"com.example.accesscontrol.exception.GlobalExceptionHandler","object_id":1039267019,"value":"com.example.accesscontrol.exception.GlobalExceptionHandler@3df1f4cb"},"static":false,"thread_id":1},{"elapsed":9.993E-5,"event":"return","id":177739,"parent_id":177738,"return_value":{"class":"org.springframework.http.ResponseEntity","object_id":1378162926,"value":"<503 SERVICE_UNAVAILABLE Service Unavailable,{status=503, timestamp=2025-08-11T17:57:03.718308400, error=Database connection failed. Please try again later.},[]>"},"thread_id":1},{"elapsed":4.2143E-4,"event":"return","id":177740,"parent_id":177737,"thread_id":1}],"classMap":[{"children":[{"children":[{"children":[{"children":[{"children":[{"labels":[],"location":"src/main/java/com/example/accesscontrol/exception/GlobalExceptionHandler.java:35","name":"handleDatabaseConnectionIssue","static":false,"type":"function"}],"location":"src/main/java/com/example/accesscontrol/exception/GlobalExceptionHandler.java","name":"GlobalExceptionHandler","static":false,"type":"class"},{"children":[{"labels":[],"location":"src/test/java/com/example/accesscontrol/exception/GlobalExceptionHandlerTest.java:83","name":"handleDatabaseIssue_dataAccessFailure","static":false,"type":"function"}],"location":"src/test/java/com/example/accesscontrol/exception/GlobalExceptionHandlerTest.java","name":"GlobalExceptionHandlerTest$DatabaseIssues","static":false,"type":"class"}],"name":"exception","type":"package"}],"name":"accesscontrol","type":"package"}],"name":"example","type":"package"}],"name":"com","type":"package"}],"version":"1.2","metadata":{"name":"Global exception handler $ database issues handle database issue _ data access failure","app":"accesscontrol","language":{"name":"java","version":"17.0.15+6-LTS","engine":"OpenJDK 64-Bit Server VM"},"client":{"name":"appmap-java","url":"https://github.com/appland/appmap-java"},"recorder":{"name":"junit","type":"tests"},"recording":{"defined_class":"com.example.accesscontrol.exception.GlobalExceptionHandlerTest$DatabaseIssues","method_id":"handleDatabaseIssue_dataAccessFailure"},"git":{"repository":"https://github.com/AlhaderMajd/accessControl-majd.git","branch":"main","commit":"92f7f24f7b2b6102a5dfc64fc47b555aa2ae103c"},"source_location":"src/test/java/com/example/accesscontrol/exception/GlobalExceptionHandlerTest.java:83","frameworks":[{"name":"JUnit","version":"5"}],"test_status":"succeeded"},"eventUpdates":{}}